### PR TITLE
PPML: BigDL Remote Attestation Service and TDX support in AttestationCLI 

### PR DESCRIPTION
**Pull request by xiangyuT**
## Description

Implement `BigDLRemoteAttestationService` to provide an attestation service which get quote and return quote verification result by REST API.
Enable `BigDLAttestationService` as attestation service, and TDX quote as quote type in `AttestationCLI`.

### CODE REVIEW

File: scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/service/BigDLRemoteAttestationService.scala [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+  final case class Result(result: Int)
+  implicit val resultFormat = jsonFormat1(Result)
+
+  val quoteVerifier = new SGXDCAPQuoteVerifierImpl()
+
+  def main(args: Array[String]): Unit = {
+
+    val logger = LogManager.getLogger(getClass)
+    case class CmdParams(serviceURL: String = "localhost",
+                          servicePort: String = "8184",
```
Comment (glorysdj) [inline scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/service/BigDLRemoteAttestationService.scala:64]: 0.0.0.0 and 9875

File: scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/service/BigDLRemoteAttestationService.scala [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+    val route: Route =
+        post {
+          path("verifyQuote") {
+            entity(as[Quote]) { quoteMsg =>
+              println(quoteMsg)
+              val verifyQuoteResult = quoteVerifier.verifyQuote(
+                Base64.getDecoder().decode(quoteMsg.quote.getBytes))
+              val res = new Result(verifyQuoteResult)
+              complete(res)
```
Comment (glorysdj) [inline scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/service/BigDLRemoteAttestationService.scala:98]: 200 for success
400 for fail

File: ppml/services/remote-attestation-service/docker/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    echo 'deb [trusted=yes arch=amd64] file:///opt/intel/sgx_debian_local_repo focal main' | tee /etc/apt/sources.list.d/intel-sgx.list && \
+    wget -qO - https://download.01.org/intel-sgx/sgx_repo/ubuntu/intel-sgx-deb.key | apt-key add - && \
+    apt-get update --fix-missing && \
+    apt-get install -y libsgx-enclave-common-dev libsgx-ae-qe3 libsgx-ae-qve libsgx-urts libsgx-dcap-ql libsgx-dcap-default-qpl libsgx-dcap-quote-verify-dev libsgx-dcap-default-qpl-dev libsgx-quote-ex-dev libsgx-uae-service libsgx-ra-network libsgx-ra-uefi libsgx-dcap-ql-dev libtdx-attest libtdx-attest-dev
+
+COPY --from=builder /opt/jdk /opt/jdk8
+COPY --from=builder /opt/spark-${SPARK_VERSION} /opt/spark-${SPARK_VERSION}
+# COPY --from=bigdl ${BIGDL_HOME} ${BIGDL_HOME}
+
+RUN cd / && wget -c https://downloads.lightbend.com/scala/2.11.8/scala-2.11.8.tgz && \
```
Comment (glorysdj) [inline ppml/services/remote-attestation-service/docker/Dockerfile:82]: do we need scala in this image?

File: ppml/services/remote-attestation-service/docker/README.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+```bash
+bash build-docker-image.sh
+```
+
+## 2. Start container
+
+### SGX
+If you want to verify SGX quote, you need to mount SGX device to the docker container.
+```bash
+export DOCKER_IMAGE=your_docker_image
```
Comment (glorysdj) [inline ppml/services/remote-attestation-service/docker/README.md:12]: we should specify the default DOCKER_IMAGE here

Comment (glorysdj) [reply ppml/services/remote-attestation-service/docker/README.md:12]: and forget export DOCKER_NAME here

File: ppml/services/remote-attestation-service/docker/Dockerfile (hunk 3 above)
Comment (xiangyuT) [reply ppml/services/remote-attestation-service/docker/Dockerfile:82]: yes, to start `BigDLRemoteAttestationService`

File: ppml/services/remote-attestation-service/docker/build-docker-image.sh [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+export IMAGE_NAME=your_image_name
+export IMAGE_VERSION=your_image_version_tag
+export JDK_URL=your_jdk_url
+
+sudo docker build \
+    --no-cache=true \
+    --build-arg http_proxy=${HTTP_PROXY} \
+    --build-arg https_proxy=${HTTPS_PROXY} \
+    --build-arg JDK_URL=${JDK_URL} \
+    -t $IMAGE_NAME:$IMAGE_VERSION -f ./Dockerfile .
```
Comment (glorysdj) [inline ppml/services/remote-attestation-service/docker/build-docker-image.sh:13]: we should specify the default IMAGE_NAME and IMAGE_VERSION

File: scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/attestation/AttestationCLI.scala [hunk 6]
```diff
@@ -42,7 +42,8 @@ object AttestationCLI {
                              asURL: String = "127.0.0.1:9000",
```
Comment (glorysdj) [inline scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/attestation/AttestationCLI.scala:42]: change to attestationServiceHost and attestationServicePort?

File: scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/attestation/BigDLRemoteAttestationService.scala [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+
+  final case class Result(result: Int)
+  implicit val resultFormat = jsonFormat1(Result)
+
+  val quoteVerifier = new SGXDCAPQuoteVerifierImpl()
+
+  def main(args: Array[String]): Unit = {
+
+    val logger = LogManager.getLogger(getClass)
+    case class CmdParams(serviceURL: String = "0.0.0.0",
```
Comment (glorysdj) [inline scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/attestation/BigDLRemoteAttestationService.scala:63]: serviceHost?

File: scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/attestation/BigDLRemoteAttestationService.scala [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+          .text("httpsKeyStorePath")
+          .action((x, c) => c.copy(httpsKeyStorePath = x))
+    }
+    val params = cmdParser.parse(args, CmdParams()).get
+
+    val route: Route =
+        post {
+          path("verifyQuote") {
+            entity(as[Quote]) { quoteMsg =>
+              println(quoteMsg)
```
Comment (glorysdj) [inline scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/attestation/BigDLRemoteAttestationService.scala:94]: use log instead

**Pull request merged**